huntsman/camera/pyro.py

```diff
@@ -23,8 +24,24 @@
 # Enable local display of remote tracebacks
 sys.excepthook = Pyro4.util.excepthook
 
+# Set up custom Pyro serialisers/de-serialisers for Astropy objects (Quantity, etc.)
+
+def astropy_to_dict(obj):
+    """Serializer function for Astropy objects using astropy.io.misc.yaml.dump()."""
+    return {"__class__": "astropy_yaml",
+            "yaml_dump": ayaml.dump(obj)}
+
+
+def dict_to_astropy(class_name, d):
+    """De-serialiser function for Astropy objects using astropy.io.misc.yaml.load()."""
+    return ayaml.load(d["yaml_dump"])
+
+SerializerBase.register_class_to_dict(u.Quantity, astropy_to_dict)
+SerializerBase.register_dict_to_class("astropy_yaml", dict_to_astropy)
+
```

Comment:
Is this custom to Pyro or just custom because of astropy quantities?  Seems like this doesn't really belong in the camera class itself.

Reply:
tBoth? This registers custom serialiser & deserialiser functions with Pyro for use with astropy quantities. That's how I was able to remove all the repetitive code was just there to convert quantities to/from floats or ints, because now the remote method calls can use quantities as arguments or return values the same as the local equivalents.